app.R

```diff
@@ -1,47 +1,97 @@
 #
-# This is a Shiny web application. You can run the application by clicking
-# the 'Run App' button above.
+# Raw data from SCBI_ForestGEO dataset
+# specifically, tree census data (2008, 2013, 2018) and tree ecology data
 #
-# Find out more about building applications with Shiny here:
-#
-#    http://shiny.rstudio.com/
+# censusJoin is a dataframe defined in MP3_eking03.Rmd
 #
 
+
 library(shiny)
+library(ggplot2)
 
 # Define UI for application that draws a histogram
 ui <- fluidPage(
+  titlePanel("Growth Rate Histograms using SCBI Census"),
+
+  sidebarLayout(
+    sidebarPanel(
+      selectInput("year_range", "Change in DBH (cm) from: ",
+                  c("2008 to 2013" = "diffdbh_0813", "2013 to 2018" = "diffdbh_1318")),
+      radioButtons("hist_type", "Histogram Type: ",
+                  c("Count" = "count", "Density" = "density"))
+    ),
 
-    # Application title
-    titlePanel("Old Faithful Geyser Data"),
-
-    # Sidebar with a slider input for number of bins
-    sidebarLayout(
-        sidebarPanel(
-            sliderInput("bins",
-                        "Number of bins:",
-                        min = 1,
-                        max = 50,
-                        value = 30)
-        ),
-
-        # Show a plot of the generated distribution
-        mainPanel(
-           plotOutput("distPlot")
-        )
+    mainPanel(
+      plotOutput("histPlot")
     )
+  )
 )
 
+
+
 # Define server logic required to draw a histogram
 server <- function(input, output) {
 
-    output$distPlot <- renderPlot({
-        # generate bins based on input$bins from ui.R
-        x    <- faithful[, 2]
-        bins <- seq(min(x), max(x), length.out = input$bins + 1)
 
-        # draw the histogram with the specified number of bins
-        hist(x, breaks = bins, col = 'darkgray', border = 'white')
+    output$histPlot <- renderPlot({
+
+      # adding DBH growth coloumns
+      censusDBH <- censusJoin %>%
```

Comment:
When I click "Run App", it says it can't find object `censusJoin`. This is b/c the code that creates this data frame doesn't exist here; rather it seems you have it in the file `MP3_eking03.Rmd`. you'll need to port over that code here, and ensure the app runs

Reply:
In the new commit, I transferred the code that defines censusJoin from MP3_eking03.Rmd to app.R and also added the other necessary packages